src/Compiler/CompilerOptions.cpp

```diff
@@ -1246,6 +1246,13 @@ void initCompilerConfig() {
     addCompilerConfig(CCM_SHARED_LIB_DEPS, extraLibs);
     addCompilerConfig(CCM_SHARED_LIB_PATH_DEPS, extraLibPaths);
   }
+
+  // Enable aggressive optimization for NNPA with -O3
+  if (OptimizationLevel == OptLevel::O3 &&
+      getTargetAccel().find("NNPA") != std::string::npos &&
+      getLLVMOption().find("enable-unsafe-fp-math") == std::string::npos) {
```

Comment:
I see, the idea is that you just check if that opt is already there. If it's there, either its `true` and there is no need to add it again, or `false` and then we don't add it also. Clever.